ext/dcos-installer/dcos_installer/cli.py

```diff
@@ -1,108 +1,163 @@
-        if do_uninstall.lower() in choices_true:
-            return
-        elif do_uninstall.lower() in choices_false:
-            sys.exit(1)
-        else:
-            log.error('Choices are [y]es or [n]o. "{}" is not a choice'.format(do_uninstall))
-
-
-def get_new_analytics(config_path=CONFIG_PATH):
-    return InstallerAnalytics(backend.get_config(config_path))
-
-
-def dispatch_action(args):
-    installer_analytics = get_new_analytics()
-
-    def preflight(args):
-        print_header("EXECUTING PREFLIGHT")
-        return action_lib.run_preflight
-
-    def postflight(args):
-        print_header("EXECUTING POSTFLIGHT")
-        return action_lib.run_postflight
-
-    def deploy(args):
-        print_header("EXECUTING DC/OS INSTALLATION")
-        return action_lib.install_dcos
-
-    def uninstall(args):
-        print_header("EXECUTING UNINSTALL")
-        tall_enough_to_ride()
-        return action_lib.uninstall_dcos
-
-    def install_prereqs(args):
-        print_header("EXECUTING INSTALL PREREQUISITES")
-        return action_lib.install_prereqs
-
-    for action in ['deploy', 'preflight', 'postflight', 'uninstall', 'install_prereqs']:
-        if getattr(args, action):
-            errors = run_loop(locals()[action](args), args)
-            if not args.disable_analytics:
-                installer_analytics.send(
-                    action=action,
-                    install_method="cli",
-                    num_errors=errors,
-                )
-            exit_code = 1 if errors > 0 else 0
-            sys.exit(exit_code)
+def setup_logger(options):
+    level = 'INFO'
+    if options.verbose:
+        level = 'DEBUG'
+    coloredlogs.install(
+        level=level,
+        level_styles={
+            'warn': {
+                'color': 'yellow'
+            },
+            'error': {
+                'color': 'red',
+                'bold': True,
+            },
+        },
+        fmt='%(message)s',
+        isatty=True
+    )
+    log.debug("Logger set to DEBUG")
+
+
+def start_installer(args=None):
+    """
+    The web based installer leverages Flask to present end-users of
+    dcos_installer with a clean web interface to configure their
+    site-based installation of DC/OS.
+    """
+    # If no args are passed to the class, then we're calling this
+    # class from another library or code so we shouldn't execute
+    # parser or anything else
+    options = parse_args(args)
+    setup_logger(options)
+    dispatch_option(options)
+    dispatch_action(options)
+
+
+def parse_args(args):
+    def print_usage():
+        return """
+Install Mesosophere's Data Center Operating System
+
+dcos_installer [-h] [-f LOG_FILE] [--hash-password HASH_PASSWORD] [-v]
+[--web | --genconf | --preflight | --deploy | --postflight | --uninstall | --validate-config | --test]
+
+Environment Settings:
+
+PORT                  Set the :port to run the web UI
```

Comment:
nit/  pre-existing: CHANNEL_NAME, BOOTSTRAP_ID shouldn't be set by end users. CHANNEL_NAME no longer exists as a configurable for this stuff.